### PR TITLE
Refactor all commands to match feature spec 

### LINKED ISSUE #58: Rename Commands
DO NOT MERGE YET until the rest of you guys settle your own PRs

### PR DESCRIPTION
**Pull request by taggyhan**
[Refactor all commands to match feature spec and introduce new room number

TC updated
Fix #58 , #52 ,#63

### CODE REVIEW

File: docs/diagrams/tracing/LogicSequenceDiagram.puml [hunk 1]
```diff
@@ -14,7 +14,7 @@ create ecp
 abp -> ecp
 abp -> ecp ++: parse(arguments)
 create ec
-ecp -> ec ++: index, editMemberDescriptor
+ecp -> ec ++: index, UpdateMemberDescriptor
```
Comment (starchypotatocode) [inline docs/diagrams/tracing/LogicSequenceDiagram.puml]: Shouldn't this be lower case like previously?

File: src/main/java/hallpointer/address/logic/commands/DeleteMemberCommand.java [hunk 2]
```diff
@@ -36,13 +37,20 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Member> lastShownList = model.getFilteredMemberList();
 
+        // Check if the index is valid
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_MEMBER_DISPLAYED_INDEX);
+            throw new CommandException(MESSAGE_INVALID_INDEX);
```
Comment (starchypotatocode) [inline src/main/java/hallpointer/address/logic/commands/DeleteMemberCommand.java:42]: May I ask why this chabge? Can't tell just from eyeballing here.

File: src/main/java/hallpointer/address/model/member/Member.java [hunk 3]
```diff
@@ -29,11 +29,25 @@ public class Member {
      */
     public Member(Name name, Telegram telegram, Room room, Set<Tag> tags) {
         requireAllNonNull(name, telegram, room, tags);
+    
+        // Validate each field using regex
```
Comment (starchypotatocode) [inline src/main/java/hallpointer/address/model/member/Member.java:33]: To my understanding this isn't needed because Parser does the validation already?

**Review by starchypotatocode — COMMENTED** [review]
LGTM, just have a few questions that would be good to answer first

File: src/main/java/hallpointer/address/logic/commands/DeleteMemberCommand.java (hunk 2 above)
Comment (taggyhan) [reply src/main/java/hallpointer/address/logic/commands/DeleteMemberCommand.java:42]: changed defintion too, trivial change

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 3 above)
Comment (taggyhan) [reply src/main/java/hallpointer/address/model/member/Member.java:33]: check room number here, parser doesnt

File: docs/diagrams/tracing/LogicSequenceDiagram.puml (hunk 1 above)
Comment (taggyhan) [reply docs/diagrams/tracing/LogicSequenceDiagram.puml]: fixed

**Pull request merged**